File: ResourceApi/Models/Pokemon.cs
using System.Collections.Generic;

namespace ResourceApi.Models;

public class Pokemon
{
    public int Id { get; set; }
    public int PokedexNumber { get; set; }
    public string Name { get; set; } = string.Empty;
    public string ImageUrl { get; set; } = string.Empty;
    public int Generation { get; set; }
    public bool IsLegendary { get; set; }
    public bool IsMythical { get; set; }

    // Add this if you still need 'BaseExperience' from the second block
    public int BaseExperience { get; set; }

    // Task 2.3.2 Requirements
    public decimal Height { get; set; } // in meters
    public decimal Weight { get; set; } // in kg

    public int HP { get; set; }
    public int Attack { get; set; }
    public int Defense { get; set; }
    public int SpecialAttack { get; set; }
    public int SpecialDefense { get; set; }
    public int Speed { get; set; }

    public ICollection<PokemonType> PokemonTypes { get; set; } = new List<PokemonType>();
}

File: Pokedexwebapplication-main/src/types/pokemon.ts
export interface Pokemon {
  id: number;
  pokedexNumber: number;
  name: string;
  imageUrl: string;
  types: string[];
  generation: number;
  isLegendary: boolean;
  isMythical: boolean;
  isCaptured?: boolean;

  // Stats (optional, for detail view)
  hp?: number;
  attack?: number;
  defense?: number;
  specialAttack?: number;
  specialDefense?: number;
  speed?: number;
  height?: number;
  weight?: number;
}

export interface PokemonListResponse {
  items: Pokemon[];
  totalCount: number;
  page: number;
  pageSize: number;
}
